module/observationSubmissions/helper.js

```diff
@@ -114,14 +114,25 @@ module.exports = class ObservationSubmissionsHelper {
 
                 let observationSubmissionsDocument = await database.models.observationSubmissions.findOne({
                     _id: observationSubmissionId,
-                    status: "completed"
                 }).lean();
 
+               
                 if (!observationSubmissionsDocument) {
-                    throw messageConstants.apiResponses.SUBMISSION_NOT_FOUND+"or"+messageConstants.apiResponses.SUBMISSION_STATUS_NOT_COMPLETE;
+                    throw messageConstants.apiResponses.SUBMISSION_NOT_FOUND;
                 }
 
-                if( observationSubmissionsDocument.referenceFrom === messageConstants.common.PROJECT ) {
+                const solutionDocument = await database.models.solutions.findOne({
+                    _id: observationSubmissionsDocument.solutionId,
+                }, { name: 1, scoringSystem : 1, description : 1, questionSequenceByEcm : 1 }).lean();
+                observationSubmissionsDocument['solutionInfo'] = solutionDocument;
+
+                const programDocument = await database.models.programs.findOne({
```

Comment:
@rakeshSgr - Use program helper for the query

Reply:
resolved